vars/updatecli.groovy

```diff
@@ -1,58 +1,95 @@
 /**
- Note: this shared pipeline is tailored to Jenkins infrastructure usage and thus set some default values which might not be desirable for yours.
+ Note: this shared pipeline is tailored to Jenkins infrastructure usage and thus sets some default values which might not be desirable for yours.
  **/
 
 def call(userConfig = [:]) {
-  def defaultConfig = [
-    action: 'diff', // Updatecli subcommand to execute
-    config: './updatecli/updatecli.d', // Config manifest used by updatecli (can be a file or a directory)
-    values: './updatecli/values.yaml', // Values file used by updatecli
-    updatecliAgentLabel: 'jnlp-linux-arm64', // replace updatecliDockerImage
-    cronTriggerExpression: '', // When specified, it enables cron trigger for the calling pipeline
-    credentialsId: 'github-app-updatecli-on-jenkins-infra', // githubApp or usernamePassword credentials id to use to get an Access Token. The corresponding populated env vars are USERNAME_VALUE & UPDATECLI_GITHUB_TOKEN
-  ]
+    def defaultConfig = [
+        action: 'diff',                         // Updatecli subcommand to execute
+        config: './updatecli/updatecli.d',      // Config manifest (file or directory) for updatecli
+        values: './updatecli/values.yaml',      // Values file used by updatecli
+        updatecliAgentLabel: 'jnlp-linux-arm64', // Label to select the Jenkins node (agent)
+        cronTriggerExpression: '',               // Enables cron trigger if specified
+        credentialsId: 'github-app-updatecli-on-jenkins-infra', // GitHub credentials
+        version: ''                               // Custom updatecli version (e.g. '0.92.0' or '0.86.0-rc.1')
+    ]
 
-  // TODO: use isInfra() to set a default githubApp credentials id for infra & for ci
+    // TODO: use isInfra() to set a default githubApp credentials id for infra & for ci
+    // Merging the 2 maps - https://blog.mrhaki.com/2010/04/groovy-goodness-adding-maps-to-map_21.html  final Map
+    finalConfig = defaultConfig << userConfig
+    final String customUpdatecliPath = "/tmp/custom_updatecli" // Factorized custom path
```

Comment:
```suggestion
    final String customUpdatecliPath = "${pwd tmp: true}/custom_updatecli"
```

- It is recommended to never use `/tmp` directly. But instead:
  - If you are in a shell context, the command `mktemp` is your friend
  - But this instruction here is NOT in a shell context: let's use a pipeline primitive to allocate a temporary directory (on the agent). Doc: https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#pwd-determine-current-directory. Example in https://github.com/jenkins-infra/pipeline-library/blob/0f75f95a5aefcb9b3d5031d4a800f0df9351e39d/vars/buildPlugin.groovy#L106
- Unneeded comment